endpoint.bal

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/io;
+import ballerina/mime;
+import ballerina/lang.'xml as xmllib;
+import ballerina/stringutils;
+
+# Eventhub client implementation.
+#
+# + config - Client configuration
+public client class Client {
+
+    private ClientEndpointConfiguration config;
+    private string API_PREFIX = EMPTY_STRING;
+    private http:Client clientEndpoint;
+
+    public function init(ClientEndpointConfiguration config) {
+        self.config = config;
+        self.API_PREFIX = TIME_OUT + config.timeout.toString() + API_VERSION + config.apiVersion;
+        self.clientEndpoint = new (HTTPS + self.config.resourceUri);
+    }
+
+    # Send a single event
+    #
+    # + eventHubPath - event hub path
+    # + data - event data
+    # + userProperties - user properties
+    # + brokerProperties - broker properties
+    # + partitionId - partition ID
+    # + publisherId - publisher ID 
+    # + return - @error if remote API is unreachable
+    remote function send(string eventHubPath, string|xml|json|byte[]|io:ReadableByteChannel|mime:Entity[] data, 
```

Comment:
Please provide another overload for passing the partition key as well. For e.g. send(EventData eventData, String partitionKey)

Reply:
Partition Key Support when publishing messages is implemented. Thank you.